lib/components/chat-input/index.tsx

```diff
@@ -1,42 +1,85 @@
 import { TextField } from "@mui/material";
 import ButtonIcon from "../button-icon";
 import IconArrowUp from "@appflowy-chat/assets/icons/arrow-up.svg?react";
+import IconStop from "@appflowy-chat/assets/icons/stop.svg?react";
 import "./index.css";
-import { ChangeEvent, FC } from "react";
+import { ChangeEvent, FormEvent, forwardRef, KeyboardEvent } from "react";
 
 interface IProps {
   value: string;
+  isGenerating: boolean;
   onChange: (e: ChangeEvent<HTMLTextAreaElement>) => void;
+  onSubmit: () => void;
+  onStop: () => void;
 }
 
-const ChatInput: FC<IProps> = ({ onChange, value }) => {
-  return (
-    <div className="w-full border border-primary-gray rounded-lg focus-within:border-accent transition-colors">
-      {/* prettier-ignore */}
-      <TextField
+const ChatInput = forwardRef<HTMLInputElement, IProps>(
+  ({ onChange, onSubmit, onStop, value, isGenerating }, ref) => {
+    function handleOnSubmit(e: FormEvent<HTMLFormElement>) {
+      e.preventDefault();
+      if (isGenerating) {
+        return;
+      }
+      onSubmit();
+    }
+    function handleOnKeyDown(e: KeyboardEvent<HTMLDivElement>) {
```

Comment:
consider debouncing the input change handler to reduce the number of state updates when typing quickly.

Reply:
Do you mean debounce the keydown change handler? If yes, it is not possible, as inside the handler we have logic that relies on the input value and isGenereting state.